bentoml/datadog_checks/bentoml/check.py

```diff
@@ -0,0 +1,57 @@
+# (C) Datadog, Inc. 2025-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from urllib.parse import urlparse, urlunparse
+
+from datadog_checks.base import OpenMetricsBaseCheckV2
+from datadog_checks.bentoml.metrics import ENDPOINT_METRICS, METRICS
+
+
+class BentomlCheck(OpenMetricsBaseCheckV2):
+    __NAMESPACE__ = 'bentoml'
+    DEFAULT_METRIC_LIMIT = 0
+
+    def __init__(self, name, init_config, instances):
+        super(BentomlCheck, self).__init__(name, init_config, instances)
+        self.openmetrics_endpoint = self.instance.get('openmetrics_endpoint')
+        self.base_url = self._extract_base_url(self.openmetrics_endpoint) if self.openmetrics_endpoint else None
+        self.tags = self.instance.get('tags', [])
+
+    def _extract_base_url(self, endpoint):
+        parsed = urlparse(endpoint)
+        path = parsed.path.rstrip('/')
+        base_path = path.rsplit('/', 1)[0] if '/' in path else ''
+        return urlunparse(parsed._replace(path=base_path or '/'))
+
+    def get_default_config(self):
+        return {
+            'openmetrics_endpoint': self.openmetrics_endpoint,
+            'metrics': [METRICS],
+        }
+
+    def check(self, instance):
+        super(BentomlCheck, self).check(instance)
+        if self.base_url:
+            self.check_health_endpoint()
+
+    def check_health_endpoint(self):
+        for endpoint_path, metric_name in ENDPOINT_METRICS.items():
```

Comment:
This looks great! Two things: 
- Add a test for endpoint metrics 
- Assert metrics using metadata

Reply:
Endpoint metrics are tested here:
https://github.com/DataDog/integrations-core/pull/21232/files#diff-9cd869502b6c4844a67fa08a1714348c5df980b8ed73053f09d95d192570a917R22

And there's a test that test them when they fail. 

I guess I could rename them to keep them consistent.

I'll add the assert with metadata